File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/Url.java
package com.noblemajesty.marvel.models.getCharacters;

import android.os.Parcel;
import android.os.Parcelable;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Url implements Parcelable
{

  private String type;
  @SerializedName("url")
  @Expose
  private String url;
  public final static Parcelable.Creator<Url> CREATOR = new Creator<Url>() {


    @SuppressWarnings({
        "unchecked"
    })
    public Url createFromParcel(Parcel in) {
      return new Url(in);
    }

    public Url[] newArray(int size) {
      return (new Url[size]);
    }

  };

  protected Url(Parcel in) {
    this.type = ((String) in.readValue((String.class.getClassLoader())));
    this.url = ((String) in.readValue((String.class.getClassLoader())));
  }

  public Url() {
  }

  public String getUrl() {
    return url;
  }

  public void writeToParcel(Parcel dest, int flags) {
    dest.writeValue(type);
    dest.writeValue(url);
  }

  public int describeContents() {
    return 0;
  }

}

File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/Url.kt
package com.noblemajesty.marvel.models.getCharacters

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Url : Parcelable {

    private val type: String
    @SerializedName("url")
    @Expose
    val url: String

    protected constructor(`in`: Parcel) {
        this.type = `in`.readValue(String::class.java.classLoader) as String
        this.url = `in`.readValue(String::class.java.classLoader) as String
    }

    constructor() {}

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(type)
        dest.writeValue(url)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<Url> = object : Parcelable.Creator<Url> {


            override fun createFromParcel(`in`: Parcel): Url {
                return Url(`in`)
            }

            override fun newArray(size: Int): Array<Url> {
                return arrayOfNulls(size)
            }

        }
    }

}